mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTableView.js

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { getSrc } from './ShowArtifactPage';
+import { getArtifactContent } from '../../../common/utils/ArtifactUtils';
+import { Table } from '@databricks/design-system';
+import { FormattedMessage } from 'react-intl';
+import Papa from 'papaparse';
+
+const ShowArtifactTableView = ({ runUuid, path, getArtifact }) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState();
+  const [data, setData] = useState();
+  const [headers, setHeaders] = useState();
+  const [text, setText] = useState();
+
+  const MAX_ROW_LENGTH = 500;
+
+  useEffect(() => {
+    resetState();
+    fetchArtifacts({ path, runUuid, getArtifact });
+  }, [runUuid, path]);
+
+  function resetState() {
+    setError();
+    setData();
+    setHeaders();
+    setText();
+    setLoading(true);
+  }
+
+  function fetchArtifacts(artifactData) {
+    const artifactLocation = getSrc(artifactData.path, artifactData.runUuid);
+    artifactData
+      .getArtifact(artifactLocation)
+      .then((artifactText) => {
+        try {
+          const result = Papa.parse(artifactText, {
+            header: true,
+            preview: MAX_ROW_LENGTH,
+          });
+          const dataPreview = result.data;
+
+          const allCellsEmpty = Object.values(dataPreview[dataPreview.length - 1]).map(
+            (value) => value === '',
+          );
+          if (allCellsEmpty.every(Boolean)) {
+            dataPreview.pop();
+          }
+
+          setLoading(false);
+          setHeaders(result.meta.fields);
+          setData(dataPreview);
+        } catch (_) {
+          setLoading(false);
+          setText(artifactText);
+        }
+      })
+      .catch((e) => {
+        setError(e);
+        setLoading(false);
+      });
+  }
+
+  if (loading) {
+    return <div className='artifact-text-view-loading'>Loading...</div>;
+  }
+  if (error) {
+    return (
+      <div className='artifact-text-view-error'>
+        Oops we couldn't load your file because of an error.
+      </div>
+    );
+  }
+
+  if (data) {
+    const columns = headers.map((f) => ({
+      title: f,
+      dataIndex: f,
+      key: f,
+      sorter: (a, b) => {
+        if (isNaN(a)) {
```

Comment:
Why are we checking whether `a` is NaN here?

Reply:
I add a check if a cell value is not a number and run a string compare function for the sorter. `papaparse` casts parsed data to their most likely data type so I'm just accounting for numbers and strings in the sorter.

Reply:
After double-checking, you are correct. Everything comes in as a string, and `null`/`undefined` becomes an empty string. I updated the sorter accordingly.